geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/data/AccumuloRasterQueryPlanner.scala

```diff
@@ -28,18 +28,35 @@ import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
 import org.opengis.feature.simple.SimpleFeatureType
 import org.opengis.filter.Filter
 
+import scala.annotation.tailrec
 import scala.collection.JavaConversions._
 import scala.util.Try
 
-// TODO: Consider adding resolutions + extent info  https://geomesa.atlassian.net/browse/GEOMESA-645
-class AccumuloRasterQueryPlanner extends Logging with IndexFilterHelpers {
-  import AccumuloRasterQueryPlanner._
+object AccumuloRasterQueryPlanner extends Logging with IndexFilterHelpers {
 
-  def getQueryPlan(rq: RasterQuery, resAndGeoHashMap: ImmutableSetMultimap[Double, Int]): Option[QueryPlan] = {
-    val availableResolutions = resAndGeoHashMap.keySet().toList.sorted
+  def improvedOverlaps(a: Envelope, b: Envelope): Boolean = a.intersects(b) && a.intersection(b).getArea != 0.0
```

Comment:
Can this relationship be expressed correctly DE-9IM?  If so, I'd suggest trying out 'relates'.  Also, this is technical enough to deserve a comment.  (I doubt any function name is going to explain this clearly.)